ext/QuadGKEnzymeExt.jl

```diff
@@ -96,6 +123,17 @@ function Enzyme.EnzymeRules.reverse(config, ofunc::Const{typeof(quadgk)}, dres::
             return ClosureVector(drev[1][1])
         end
         _df.f
+    elseif f isa MixedDuplicated
+        segbuf = cache[1]
+        fwd, rev = Enzyme.autodiff_thunk(ReverseSplitNoPrimal, Const{typeof(call)}, Active, typeof(f), Const{T})
+        _df, _ = quadgk(map(x->x.val, segs)...; kws..., eval_segbuf=segbuf, maxevals=0, norm=f->0) do x
+            fshadow = Ref(Enzyme.make_zero(f.val))
+            tape, prim, shad = fwd(Const(call), MixedDuplicated(f.val, fshadow), Const(x))
+            drev = rev(Const(call), f, Const(x), dres.val[1], tape)
```

Comment:
Can you explain why it's OK to use `f` and not `MixedDuplicated(f.val, fshadow)` in the reverse pass here? Both alternatives give correct results, but I can't wrap my head around why. When adapting the code for Duplicated I have to use `Duplicated(f.val, fshadow)` in both `fwd` and `rev`, otherwise I get incorrect results.

Reply:
A typo lol

Reply:
Sorry this was hacked on my phone during a cubs game

Reply:
that's nuts :baseball: :exploding_head: 
...and leaves me even more confused about getting correct results on every test case I've tried so far, but glad to know my intuition was right